runtime/compiler/runtime/J9CodeCache.cpp

```diff
@@ -763,3 +806,50 @@ extern "C"
       }
 
    }
+
+
+int32_t
+J9::CodeCache::disclaim(TR::CodeCacheManager *manager, bool canDisclaimOnSwap)
+   {
+   int32_t disclaimDone = 0;
+
+#ifdef LINUX
+   bool trace = TR::Options::getCmdLineOptions()->getVerboseOption(TR_VerbosePerformance);
+   uint8_t *disclaim_start = _coldCodeAlloc;
+   size_t pageSize = DISCLAIM_PAGE_SIZE;
+   size_t round = pageSize - 1;
+   disclaim_start = (uint8_t *)(((size_t)(disclaim_start + round)) & ~round);
+   size_t disclaim_size = pageSize * ((_coldCodeAllocBase - disclaim_start)/pageSize);
+
+   if (trace)
+      {
+      size_t warm_size = _warmCodeAlloc - _segment->segmentBase() + sizeof(this);
+      size_t cold_size = _coldCodeAllocBase - _coldCodeAlloc;
+
+      TR_VerboseLog::writeLine(TR_Vlog_PERF, "Disclaiming cold code cache %p : coldStart=%p coldBase=%p warm_size=%zuB cold_size=%zuB cold_size/warm_size=%5.2f%%\n",
+                               this, _coldCodeAlloc, _coldCodeAllocBase,
+                               warm_size, cold_size, cold_size * 100.0/warm_size);
```

Comment:
Unlikely, but warm_size could be 0 in theory sometimes.

Reply:
There is at least one instruction (to jump to the cold cache) in the warm code. But I changed the code to print `cold_size/(cold_size+warm_size) `since it seems to be more useful.